src/ceph-volume/ceph_volume/util/lsmdisk.py

```diff
@@ -0,0 +1,203 @@
+        """Query a disks IDENT LED state to discover when it is On, Off or Unknown (str)"""
+        led_state = self.led_status
+        if led_state == 1:
+            return "Unsupported"
+        if led_state & lsm.Disk.LED_STATUS_IDENT_ON == lsm.Disk.LED_STATUS_IDENT_ON:
+            return "On"
+        elif led_state & lsm.Disk.LED_STATUS_IDENT_OFF == lsm.Disk.LED_STATUS_IDENT_OFF:
+            return "Off"
+        elif led_state & lsm.Disk.LED_STATUS_IDENT_UNKNOWN == lsm.Disk.LED_STATUS_IDENT_UNKNOWN:
+            return "Unknown"
+        
+        return "Unsupported"
+
+    @property
+    def led_fault_state(self):
+        """Query a disks FAULT LED state to discover when it is On, Off or Unknown (str)"""
+        led_state = self.led_status
+        if led_state == 1:
+            return "Unsupported"
+        if led_state & lsm.Disk.LED_STATUS_FAULT_ON == lsm.Disk.LED_STATUS_FAULT_ON:
+            return "On"
+        elif led_state & lsm.Disk.LED_STATUS_FAULT_OFF == lsm.Disk.LED_STATUS_FAULT_OFF:
+            return "Off"
+        elif led_state & lsm.Disk.LED_STATUS_FAULT_UNKNOWN == lsm.Disk.LED_STATUS_FAULT_UNKNOWN:
+            return "Unknown"
+        
+        return "Unsupported"
+
+    @property
+    def led_ident_support(self):
+        """Query the LED state to determine IDENT support: Unknown, Supported, Unsupported (str)"""
+        led_state = self.led_status
+        if led_state == 1:
+            return "Unknown"
+
+        ident_states = (
+            lsm.Disk.LED_STATUS_IDENT_ON + 
+            lsm.Disk.LED_STATUS_IDENT_OFF + 
+            lsm.Disk.LED_STATUS_IDENT_UNKNOWN
+        )
+
+        if (led_state & ident_states) == 0:
+            return "Unsupported"
+        
+        return "Supported"
+
+    @property
+    def led_fault_support(self):
+        """Query the LED state to determine FAULT support: Unknown, Supported, Unsupported (str)"""
+        led_state = self.led_status
+        if led_state == 1:
+            return "Unknown"
+
+        fail_states = (
+            lsm.Disk.LED_STATUS_FAULT_ON + 
+            lsm.Disk.LED_STATUS_FAULT_OFF + 
+            lsm.Disk.LED_STATUS_FAULT_UNKNOWN
+        )
+
+        if led_state & fail_states == 0:
+            return "Unsupported"
+
+        return "Supported"
+
+    @property
+    def health(self):
+        """Determine the health of the disk from LSM : Unknown, Fail, Warn or Good (str)"""
+        health_map = {
+           -1: "Unknown",
+            0: "Fail",
+            1: "Warn",
+            2: "Good",
+        }
+        _health_int = self._query_lsm('health_status_get', self.dev_path)
+        if _health_int is None:
+            _health_int = -1
+        return health_map.get(_health_int, "Unknown")
+
+    @property
+    def transport(self):
+        """Translate a disks link type to a human readable format (str)"""
+        _link_type = self._query_lsm('link_type_get', self.dev_path)
+
+        if _link_type is not None:
```

Comment:
if you create an lsmDisk, and ignore the lsm_available flag and try to access the transport attribute, the query to lsm will return None. This block is there to catch this scenario where you try to get the attribute when lsm is not available.

Reply:
Right, I might be missing somehting but if `_link_type` is `None` (and we get rid of the if block) the get call to the transport map will be `_transport_map.get(None, "Unknown")` which returns `"Unkown"`

Reply:
agree - I'm looking at this from the class side. If you create an instance, and try to access the attributes without checking lsm_available you should not be greeted with a NameError. By removing the if block, that's what you get, since the lsm globals have not been defined via the import.
All the other references for the lsm_Disk globals are "protected" in the same way. I don't know how the LSMDisk class is going to be used in future - so I'm trying to play safe (by nurturing a healthy sense of paranoia)!

Reply:
refactor has removed this issue